open-api/rest-catalog-open-api.yaml

```diff
@@ -549,28 +549,57 @@ paths:
     post:
       tags:
         - Catalog API
-      summary: Returns either a list of `PlanTask`, a list of `FileScanTask`. If planning is not complete returns a `plan-id`.
+      summary: Submit a scan for planning
       description:
-        Prepares either a list of `PlanTask` that can be used to distribute table scan planning, or a list of `FileScanTask`, based on a set of table scan criteria
-        such as selected columns, filters, snapshot range, case sensitivity, etc.
-        In the event that the plan tasks or file scan tasks are not ready to be served, the service will return a `plan-id`,
-        which can be used as input in the `GetTasksStatus`.
-        
-        Requires that the client specifies only a `snapshot-id` for a snapshot scan, or for performing incremental scans only provide 
-        a `start-snapshot-id` and an `end-snapshot-id`. 
-      
-        Each `PlanTask`, can be used as input in the `RetrieveTasks` API
-        to request a subset of file scan tasks in a table scan.
-        This mechanism allows clients to distribute and parallelize the entire table scan planning process.
-      operationId: planTable
+        Submits a scan for server-side planning.
+
+
+        Point-in-time scans are planned by passing snapshot-id to identify the
+        table snapshot to scan. Incremental scans are planned by passing both
+        start-snapshot-id and end-snapshot-id. Requests that include both point
+        in time config properties and incremental config properties are
+        invalid. If the request does not include either incremental or
+        point-in-time config properties, scan planning should produce a
+        point-in-time scan of the latest snapshot in the table's main branch.
+
+
+        Responses must include a valid status
+
+        - When "completed" the planning operation has produced plan-tasks and
```

Comment:
@rdblue What is the benefit for having this additional `completed` enum? I thought the presence of the plan tasks, file scan task is enough to signal completion. Is it just that its cleaner to have this additional state as a client will polling on this enum state.

Reply:
> the planning operation has produced plan-tasks and file-scan-tasks

For this statement, is it a requirement that the client must return both these types of tasks?

I thought that its optional for the client to return only plan tasks, return only file scan tasks, or return both.

Reply:
As long as we are sending a status, we may as well use it to be more clear about the state. As you noted earlier, this allows us to make statements like "it is not necessary to cancel a plan-id that has returned a 'completed'" state. Here, I think having it avoids the need to tell implementers what the _implied_ status is when plan-tasks and/or file-scan-tasks are included. Instead it is obvious that `completed` means that the planning is done.

Reply:
Makes sense